membership_delegated_partner/tests/test_membership_delegate.py

```diff
@@ -97,23 +98,14 @@ def test_02_change_delegated_member(self):
 
     def test_03_refund_invoice_delegated_partner(self):
         """ A refund should inherit the delegated partner in the invoice """
-        invoice = self.env["account.move"].create(
-            {
-                "name": "Test Customer Invoice",
-                "move_type": "out_invoice",
-                "partner_id": self.partner1.id,  # Invoicing partner
-                "delegated_member_id": self.partner2.id,  # Delegate membership to
-            }
-        )
-        self.env["account.move.line"].create(
-            {
-                "move_id": invoice.id,
-                "name": "Membership for delegate member",
-                "account_id": self.account.id,
-                "product_id": self.product.id,
-                "price_unit": 1.0,
-            }
-        )
+        move_form = Form(self.env['account.move'].with_context(default_move_type='out_invoice'))
+        move_form.partner_id = self.partner1
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product
+            line_form.account_id = self.account
+            line_form.price_unit = 1.0
+        invoice = move_form.save()
+        invoice.write({"delegated_member_id": self.partner2.id})
```

Comment:
Why not doing this on the Form?

Reply:
I was waiting for that :smiley: Because the filed is not in the Form, so not writeable there

Reply:
But IMHO, after this, tests should be enhanced to be done with Form as some values cannot be initialized.

Reply:
hehe, OK, but maybe it's because you are selecting the incorrect view? I see that the field is added in the proper one:

https://github.com/OCA/vertical-association/blob/12.0/membership_delegated_partner/views/account_invoice.xml#L10